ISSUES.md

```diff
@@ -0,0 +1,223 @@
+
+8. Before reporting an Issue, first, always try to update your environment to the latest official version of this library. We have no resources to go and debug older revisions, which could easily have bugs that have been fixed in the latest released version.
+
+   We understand that this is not always possible, especially when APIs change, in which case file an Issue against the highest library version your environment can support.
+   
+   Of course, if you upgrade the library, always retest that the problem is still there.
+
+9. Please do not ask us to reproduce an Issue with your custom data, since we don't have it. So, either you should use some existing dataset supported by HF datasets or you need to supply a code that generates a small sample on the fly, or some another quick and simple way to get it.
+
+   Please do not send us any non-public domain data that may require a license or a permission to be used.
+
+10. Do not tag multiple developers on the Issue unless you know this is expected, either because you asked them and they gave you an explicit permission to tag them or the Issue template instructs you to do so. 
+
+   The "who to tag for what domain" part of the Issue template is there to help users direct their questions to the right developers who are designated maintainers of project's specific domains. They can then decide at their own discretion to tag other developers if they feel it'd help move the Issue forward.
+   
+   We currently don't have a triage service and we trust your capacity to identify the right domain and thus the persons to tag in your Issue. If you are not sure, please use the forums to ask for guidance.
+   
+   When in doubt, err on the side of not tagging a given person. If you tag multiple people out of context or permission don't be surprised if you get no response at all. Please remember that every time you tag someone, they get a notification and you're taking their time without their permission. Please be sensitive to that.
+
+   If you got helped by one of the developers in the past please don't tag them in future Issues, unless they are listed in the Issue template for the domain you are asking about or that developer gave you an explicit permission to tag them in future Issues.
+   
+   If you see a certain developer doing multiple and/or recent commits into a specific area of the project that you feel is relevant to your Issue, it is not a good reason to tag them. Various developers may be fixing things that prevent them from moving forward, but often their work is focused on a totally different domain. And while they may or may not know how to help you with the problem at hand, it would benefit the whole community much more if they focus on the domain of their unique expertise.
```

Comment:
I somewhat disagree with that statement. If the issue respects the rest of this document, then it is well structured, the problem is well explained, and a member of the team that recently worked on that area of the project will most likely be the best suited to answer the issue. It's a very fast moving project so members that worked on an aspect of the project one year ago are not experts on that area anymore, but users that recently edited/worked on that area most likely have the best understanding of that part.

Reply:
But that's why the Issue/PR templates have designated domain "owners" - those are the people who know the domain - their role could be to dispatch the issue to the person they know is most familiar with the issue at hand. Asking users to figure out who to tag is just unfair to the user and is asking too much, IMHO. 

I like how some projects have a triage process - e.g. pytorch - the user is not requested to tag anybody. Whoever is doing the triage - and they switch labels to handle this process - tags one or more developers who they know would be the most relevant developer to take care of the issue at hand. Surely, pytorch is much bigger and has many more issues and developers to handle such process, but what I like is that the project takes care of this on behalf of users. 

So perhaps the most user-friendly solution is just that - don't tag - we will tag for you. 

This is just a suggestion as everything else I have written in this document. Please feel free to edit/remove/rewrite as you feel fit.

somewhat related - if user-tags-developers is a thing - then using labels would make it easier to find things for both developers and users. So users could hit a label of their interest and see which developer does most work in this area as of recent.

Reply:
I wasn't thinking that we had to specify "You should look who last edited these files", but rather that we don't need to put in the document "Do not look at who last edited these files to tag them". In any case, I agree with your points. Tagging the domain owner so that he can tag others if they are more knowledgeable than the owner is good enough for me.